--- /tmp/armv8m_mpu.h.orig	2026-01-13 17:49:25.801105239 +0000
+++ envs/m33-an524/src/platform/m-profile/armv8m_mpu.h	2026-01-13 15:11:08.455800010 +0000
@@ -1,9 +1,3 @@
-/******************************************************************************
- * @file     mpu_armv8.h
- * @brief    CMSIS MPU API for Armv8-M and Armv8.1-M MPU
- * @version  V5.9.0
- * @date     11. April 2023
- ******************************************************************************/
 /*
  * Copyright (c) 2017-2022 Arm Limited. All rights reserved.
  *
@@ -22,86 +16,103 @@
  * limitations under the License.
  */
 
-#if   defined ( __ICCARM__ )
-  #pragma system_include         /* treat file as system include file for MISRA check */
-#elif defined (__clang__)
-  #pragma clang system_header    /* treat file as system include file */
-#endif
+/*
+ * CMSIS-Core(M) MPU API for Armv8-M and Armv8.1-M MPU
+ */
 
 #ifndef ARM_MPU_ARMV8_H
 #define ARM_MPU_ARMV8_H
 
+#if defined(__ICCARM__)
+#pragma system_include /* treat file as system include file for MISRA check */
+#elif defined(__clang__)
+#pragma clang system_header /* treat file as system include file */
+#endif
+
 /** \brief Attribute for device memory (outer only) */
-#define ARM_MPU_ATTR_DEVICE                           ( 0U )
+#define ARM_MPU_ATTR_DEVICE (0U)
 
 /** \brief Attribute for non-cacheable, normal memory */
-#define ARM_MPU_ATTR_NON_CACHEABLE                    ( 4U )
+#define ARM_MPU_ATTR_NON_CACHEABLE (4U)
 
 /** \brief Attribute for Normal memory, Outer and Inner cacheability.
-* \param NT Non-Transient: Set to 1 for Non-transient data. Set to 0 for Transient data.
-* \param WB Write-Back: Set to 1 to use a Write-Back policy. Set to 0 to use a Write-Through policy.
-* \param RA Read Allocation: Set to 1 to enable cache allocation on read miss. Set to 0 to disable cache allocation on read miss.
-* \param WA Write Allocation: Set to 1 to enable cache allocation on write miss. Set to 0 to disable cache allocation on write miss.
-*/
-#define ARM_MPU_ATTR_MEMORY_(NT, WB, RA, WA) \
-  ((((NT) & 1U) << 3U) | (((WB) & 1U) << 2U) | (((RA) & 1U) << 1U) | ((WA) & 1U))
+ * \param NT Non-Transient: Set to 1 for Non-transient data. Set to 0 for
+ * Transient data.
+ * \param WB Write-Back: Set to 1 to use a Write-Back policy. Set to 0 to use a
+ * Write-Through policy.
+ * \param RA Read Allocation: Set to 1 to enable cache allocation on read miss.
+ * Set to 0 to disable cache allocation on read miss.
+ * \param WA Write Allocation: Set to 1 to enable cache allocation on write
+ * miss. Set to 0 to disable cache allocation on write miss.
+ */
+#define ARM_MPU_ATTR_MEMORY_(NT, WB, RA, WA)                         \
+  ((((NT) & 1U) << 3U) | (((WB) & 1U) << 2U) | (((RA) & 1U) << 1U) | \
+   ((WA) & 1U))
 
-/** \brief Device memory type non Gathering, non Re-ordering, non Early Write Acknowledgement */
+/** \brief Device memory type non Gathering, non Re-ordering, non Early Write
+ * Acknowledgement */
 #define ARM_MPU_ATTR_DEVICE_nGnRnE (0U)
 
-/** \brief Device memory type non Gathering, non Re-ordering, Early Write Acknowledgement */
-#define ARM_MPU_ATTR_DEVICE_nGnRE  (1U)
-
-/** \brief Device memory type non Gathering, Re-ordering, Early Write Acknowledgement */
-#define ARM_MPU_ATTR_DEVICE_nGRE   (2U)
-
-/** \brief Device memory type Gathering, Re-ordering, Early Write Acknowledgement */
-#define ARM_MPU_ATTR_DEVICE_GRE    (3U)
+/** \brief Device memory type non Gathering, non Re-ordering, Early Write
+ * Acknowledgement */
+#define ARM_MPU_ATTR_DEVICE_nGnRE (1U)
+
+/** \brief Device memory type non Gathering, Re-ordering, Early Write
+ * Acknowledgement */
+#define ARM_MPU_ATTR_DEVICE_nGRE (2U)
+
+/** \brief Device memory type Gathering, Re-ordering, Early Write
+ * Acknowledgement */
+#define ARM_MPU_ATTR_DEVICE_GRE (3U)
 
 /** \brief Normal memory outer-cacheable and inner-cacheable attributes
-* WT = Write Through, WB = Write Back, TR = Transient, RA = Read-Allocate, WA = Write Allocate
-*/
+ * WT = Write Through, WB = Write Back, TR = Transient, RA = Read-Allocate, WA =
+ * Write Allocate
+ */
 #define MPU_ATTR_NORMAL_OUTER_NON_CACHEABLE (0b0100)
-#define MPU_ATTR_NORMAL_OUTER_WT_TR_RA      (0b0010)
-#define MPU_ATTR_NORMAL_OUTER_WT_TR_WA      (0b0001)
-#define MPU_ATTR_NORMAL_OUTER_WT_TR_RA_WA   (0b0011)
-#define MPU_ATTR_NORMAL_OUTER_WT_RA         (0b1010)
-#define MPU_ATTR_NORMAL_OUTER_WT_WA         (0b1001)
-#define MPU_ATTR_NORMAL_OUTER_WT_RA_WA      (0b1011)
-#define MPU_ATTR_NORMAL_OUTER_WB_TR_RA      (0b0101)
-#define MPU_ATTR_NORMAL_OUTER_WB_TR_WA      (0b0110)
-#define MPU_ATTR_NORMAL_OUTER_WB_TR_RA_WA   (0b0111)
-#define MPU_ATTR_NORMAL_OUTER_WB_RA         (0b1101)
-#define MPU_ATTR_NORMAL_OUTER_WB_WA         (0b1110)
-#define MPU_ATTR_NORMAL_OUTER_WB_RA_WA      (0b1111)
+#define MPU_ATTR_NORMAL_OUTER_WT_TR_RA (0b0010)
+#define MPU_ATTR_NORMAL_OUTER_WT_TR_WA (0b0001)
+#define MPU_ATTR_NORMAL_OUTER_WT_TR_RA_WA (0b0011)
+#define MPU_ATTR_NORMAL_OUTER_WT_RA (0b1010)
+#define MPU_ATTR_NORMAL_OUTER_WT_WA (0b1001)
+#define MPU_ATTR_NORMAL_OUTER_WT_RA_WA (0b1011)
+#define MPU_ATTR_NORMAL_OUTER_WB_TR_RA (0b0110)
+#define MPU_ATTR_NORMAL_OUTER_WB_TR_WA (0b0101)
+#define MPU_ATTR_NORMAL_OUTER_WB_TR_RA_WA (0b0111)
+#define MPU_ATTR_NORMAL_OUTER_WB_RA (0b1110)
+#define MPU_ATTR_NORMAL_OUTER_WB_WA (0b1101)
+#define MPU_ATTR_NORMAL_OUTER_WB_RA_WA (0b1111)
 #define MPU_ATTR_NORMAL_INNER_NON_CACHEABLE (0b0100)
-#define MPU_ATTR_NORMAL_INNER_WT_TR_RA      (0b0010)
-#define MPU_ATTR_NORMAL_INNER_WT_TR_WA      (0b0001)
-#define MPU_ATTR_NORMAL_INNER_WT_TR_RA_WA   (0b0011)
-#define MPU_ATTR_NORMAL_INNER_WT_RA         (0b1010)
-#define MPU_ATTR_NORMAL_INNER_WT_WA         (0b1001)
-#define MPU_ATTR_NORMAL_INNER_WT_RA_WA      (0b1011)
-#define MPU_ATTR_NORMAL_INNER_WB_TR_RA      (0b0101)
-#define MPU_ATTR_NORMAL_INNER_WB_TR_WA      (0b0110)
-#define MPU_ATTR_NORMAL_INNER_WB_TR_RA_WA   (0b0111)
-#define MPU_ATTR_NORMAL_INNER_WB_RA         (0b1101)
-#define MPU_ATTR_NORMAL_INNER_WB_WA         (0b1110)
-#define MPU_ATTR_NORMAL_INNER_WB_RA_WA      (0b1111)
+#define MPU_ATTR_NORMAL_INNER_WT_TR_RA (0b0010)
+#define MPU_ATTR_NORMAL_INNER_WT_TR_WA (0b0001)
+#define MPU_ATTR_NORMAL_INNER_WT_TR_RA_WA (0b0011)
+#define MPU_ATTR_NORMAL_INNER_WT_RA (0b1010)
+#define MPU_ATTR_NORMAL_INNER_WT_WA (0b1001)
+#define MPU_ATTR_NORMAL_INNER_WT_RA_WA (0b1011)
+#define MPU_ATTR_NORMAL_INNER_WB_TR_RA (0b0110)
+#define MPU_ATTR_NORMAL_INNER_WB_TR_WA (0b0101)
+#define MPU_ATTR_NORMAL_INNER_WB_TR_RA_WA (0b0111)
+#define MPU_ATTR_NORMAL_INNER_WB_RA (0b1110)
+#define MPU_ATTR_NORMAL_INNER_WB_WA (0b1101)
+#define MPU_ATTR_NORMAL_INNER_WB_RA_WA (0b1111)
 
 /** \brief Memory Attribute
-* \param O Outer memory attributes
-* \param I O == ARM_MPU_ATTR_DEVICE: Device memory attributes, else: Inner memory attributes
-*/
-#define ARM_MPU_ATTR(O, I) ((((O) & 0xFU) << 4U) | ((((O) & 0xFU) != 0U) ? ((I) & 0xFU) : (((I) & 0x3U) << 2U)))
+ * \param O Outer memory attributes
+ * \param I O == ARM_MPU_ATTR_DEVICE: Device memory attributes, else: Inner
+ * memory attributes
+ */
+#define ARM_MPU_ATTR(O, I) \
+  ((((O) & 0xFU) << 4U) |  \
+   ((((O) & 0xFU) != 0U) ? ((I) & 0xFU) : (((I) & 0x3U) << 2U)))
 
 /* \brief Specifies MAIR_ATTR number */
-#define MAIR_ATTR(x)						((x > 7 || x < 0) ? 0 : x)
+#define MAIR_ATTR(x) ((x > 7 || x < 0) ? 0 : x)
 
 /**
  * Shareability
  */
 /** \brief Normal memory, non-shareable  */
-#define ARM_MPU_SH_NON   (0U)
+#define ARM_MPU_SH_NON (0U)
 
 /** \brief Normal memory, outer shareable  */
 #define ARM_MPU_SH_OUTER (2U)
@@ -111,7 +122,8 @@
 
 /**
  * Access permissions
- * AP = Access permission, RO = Read-only, RW = Read/Write, NP = Any privilege, PO = Privileged code only
+ * AP = Access permission, RO = Read-only, RW = Read/Write, NP = Any privilege,
+ * PO = Privileged code only
  */
 /** \brief Normal memory, read/write */
 #define ARM_MPU_AP_RW (0U)
@@ -129,75 +141,82 @@
  * Execute-never
  * XN = Execute-never, EX = Executable
  */
-/** \brief Normal memory, Execution only permitted if read permitted */
+/** \brief Normal memory, Execution not permitted */
 #define ARM_MPU_XN (1U)
 
 /** \brief Normal memory, Execution only permitted if read permitted */
 #define ARM_MPU_EX (0U)
 
 /** \brief Memory access permissions
-* \param RO Read-Only: Set to 1 for read-only memory. Set to 0 for a read/write memory.
-* \param NP Non-Privileged: Set to 1 for non-privileged memory. Set to 0 for privileged memory.
-*/
+ * \param RO Read-Only: Set to 1 for read-only memory. Set to 0 for a read/write
+ * memory.
+ * \param NP Non-Privileged: Set to 1 for non-privileged memory. Set to 0 for
+ * privileged memory.
+ */
 #define ARM_MPU_AP_(RO, NP) ((((RO) & 1U) << 1U) | ((NP) & 1U))
 
 /** \brief Region Base Address Register value
-* \param BASE The base address bits [31:5] of a memory region. The value is zero extended. Effective address gets 32 byte aligned.
-* \param SH Defines the Shareability domain for this memory region.
-* \param RO Read-Only: Set to 1 for a read-only memory region. Set to 0 for a read/write memory region.
-* \param NP Non-Privileged: Set to 1 for a non-privileged memory region. Set to 0 for privileged memory region.
-* \param XN eXecute Never: Set to 1 for a non-executable memory region. Set to 0 for an executable memory region.
-*/
-#define ARM_MPU_RBAR(BASE, SH, RO, NP, XN) \
-  (((BASE) & MPU_RBAR_BASE_Msk) | \
-  (((SH) << MPU_RBAR_SH_Pos) & MPU_RBAR_SH_Msk) | \
-  ((ARM_MPU_AP_(RO, NP) << MPU_RBAR_AP_Pos) & MPU_RBAR_AP_Msk) | \
-  (((XN) << MPU_RBAR_XN_Pos) & MPU_RBAR_XN_Msk))
+ * \param BASE The base address bits [31:5] of a memory region. The value is
+ * zero extended. Effective address gets 32 byte aligned.
+ * \param SH Defines the Shareability domain for this memory region.
+ * \param RO Read-Only: Set to 1 for a read-only memory region. Set to 0 for a
+ * read/write memory region.
+ * \param NP Non-Privileged: Set to 1 for a non-privileged memory region. Set to
+ * 0 for privileged memory region.
+ * \param XN eXecute Never: Set to 1 for a non-executable memory region. Set to
+ * 0 for an executable memory region.
+ */
+#define ARM_MPU_RBAR(BASE, SH, RO, NP, XN)                        \
+  (((BASE) & MPU_RBAR_BASE_Msk) |                                 \
+   (((SH) << MPU_RBAR_SH_Pos) & MPU_RBAR_SH_Msk) |                \
+   ((ARM_MPU_AP_(RO, NP) << MPU_RBAR_AP_Pos) & MPU_RBAR_AP_Msk) | \
+   (((XN) << MPU_RBAR_XN_Pos) & MPU_RBAR_XN_Msk))
 
 /** \brief Region Limit Address Register value
-* \param LIMIT The limit address bits [31:5] for this memory region. The value is one extended.
-* \param IDX The attribute index to be associated with this memory region.
-*/
-#define ARM_MPU_RLAR(LIMIT, IDX) \
-  (((LIMIT) & MPU_RLAR_LIMIT_Msk) | \
-  (((IDX) << MPU_RLAR_AttrIndx_Pos) & MPU_RLAR_AttrIndx_Msk) | \
-  (MPU_RLAR_EN_Msk))
+ * \param LIMIT The limit address bits [31:5] for this memory region. The value
+ * is one extended.
+ * \param IDX The attribute index to be associated with this memory region.
+ */
+#define ARM_MPU_RLAR(LIMIT, IDX)                                \
+  (((LIMIT) & MPU_RLAR_LIMIT_Msk) |                             \
+   (((IDX) << MPU_RLAR_AttrIndx_Pos) & MPU_RLAR_AttrIndx_Msk) | \
+   (MPU_RLAR_EN_Msk))
 
 #if defined(MPU_RLAR_PXN_Pos)
-  
+
 /** \brief Region Limit Address Register with PXN value
-* \param LIMIT The limit address bits [31:5] for this memory region. The value is one extended.
-* \param PXN Privileged execute never. Defines whether code can be executed from this privileged region.
-* \param IDX The attribute index to be associated with this memory region.
-*/
-#define ARM_MPU_RLAR_PXN(LIMIT, PXN, IDX) \
-  (((LIMIT) & MPU_RLAR_LIMIT_Msk) | \
-  (((PXN) << MPU_RLAR_PXN_Pos) & MPU_RLAR_PXN_Msk) | \
-  (((IDX) << MPU_RLAR_AttrIndx_Pos) & MPU_RLAR_AttrIndx_Msk) | \
-  (MPU_RLAR_EN_Msk))
-  
-#endif
+ * \param LIMIT The limit address bits [31:5] for this memory region. The value
+ * is one extended.
+ * \param PXN Privileged execute never. Defines whether code can be executed
+ * from this privileged region.
+ * \param IDX The attribute index to be associated with this memory region.
+ */
+#define ARM_MPU_RLAR_PXN(LIMIT, PXN, IDX)                       \
+  (((LIMIT) & MPU_RLAR_LIMIT_Msk) |                             \
+   (((PXN) << MPU_RLAR_PXN_Pos) & MPU_RLAR_PXN_Msk) |           \
+   (((IDX) << MPU_RLAR_AttrIndx_Pos) & MPU_RLAR_AttrIndx_Msk) | \
+   (MPU_RLAR_EN_Msk))
+
+#endif /* MPU_RLAR_PXN_Pos */
 
 /**
-* Struct for a single MPU Region
-*/
-typedef struct {
-  uint32_t RBAR;                   /*!< Region Base Address Register value */
-  uint32_t RLAR;                   /*!< Region Limit Address Register value */
+ * Struct for a single MPU Region
+ */
+typedef struct
+{
+  uint32_t RBAR; /*!< Region Base Address Register value */
+  uint32_t RLAR; /*!< Region Limit Address Register value */
 } ARM_MPU_Region_t;
 
 /**
   \brief  Read MPU Type Register
   \return Number of MPU regions
 */
-__STATIC_INLINE uint32_t ARM_MPU_TYPE()
-{
-  return ((MPU->TYPE) >> 8);
-}
-    
+__STATIC_INLINE uint32_t ARM_MPU_TYPE(void) { return ((MPU->TYPE) >> 8); }
+
 /** Enable the MPU.
-* \param MPU_Control Default access permissions for unconfigured regions.
-*/
+ * \param MPU_Control Default access permissions for unconfigured regions.
+ */
 __STATIC_INLINE void ARM_MPU_Enable(uint32_t MPU_Control)
 {
   __DMB();
@@ -210,22 +229,22 @@
 }
 
 /** Disable the MPU.
-*/
+ */
 __STATIC_INLINE void ARM_MPU_Disable(void)
 {
   __DMB();
 #ifdef SCB_SHCSR_MEMFAULTENA_Msk
   SCB->SHCSR &= ~SCB_SHCSR_MEMFAULTENA_Msk;
 #endif
-  MPU->CTRL  &= ~MPU_CTRL_ENABLE_Msk;
+  MPU->CTRL &= ~MPU_CTRL_ENABLE_Msk;
   __DSB();
   __ISB();
 }
 
 #ifdef MPU_NS
 /** Enable the Non-secure MPU.
-* \param MPU_Control Default access permissions for unconfigured regions.
-*/
+ * \param MPU_Control Default access permissions for unconfigured regions.
+ */
 __STATIC_INLINE void ARM_MPU_Enable_NS(uint32_t MPU_Control)
 {
   __DMB();
@@ -238,42 +257,43 @@
 }
 
 /** Disable the Non-secure MPU.
-*/
+ */
 __STATIC_INLINE void ARM_MPU_Disable_NS(void)
 {
   __DMB();
 #ifdef SCB_SHCSR_MEMFAULTENA_Msk
   SCB_NS->SHCSR &= ~SCB_SHCSR_MEMFAULTENA_Msk;
 #endif
-  MPU_NS->CTRL  &= ~MPU_CTRL_ENABLE_Msk;
+  MPU_NS->CTRL &= ~MPU_CTRL_ENABLE_Msk;
   __DSB();
   __ISB();
 }
-#endif
+#endif /* MPU_NS */
 
 /** Set the memory attribute encoding to the given MPU.
-* \param mpu Pointer to the MPU to be configured.
-* \param idx The attribute index to be set [0-7]
-* \param attr The attribute value to be set.
-*/
-__STATIC_INLINE void ARM_MPU_SetMemAttrEx(MPU_Type* mpu, uint8_t idx, uint8_t attr)
+ * \param mpu Pointer to the MPU to be configured.
+ * \param idx The attribute index to be set [0-7]
+ * \param attr The attribute value to be set.
+ */
+__STATIC_INLINE void ARM_MPU_SetMemAttrEx(MPU_Type *mpu, uint8_t idx,
+                                          uint8_t attr)
 {
   const uint8_t reg = idx / 4U;
   const uint32_t pos = ((idx % 4U) * 8U);
   const uint32_t mask = 0xFFU << pos;
-  const uint32_t val = (uint32_t)attr << pos;
-  
-  if (reg >= (sizeof(mpu->MAIR) / sizeof(mpu->MAIR[0]))) {
-    return; // invalid index
+
+  if (reg >= (sizeof(mpu->MAIR) / sizeof(mpu->MAIR[0])))
+  {
+    return;  // invalid index
   }
-  
-  mpu->MAIR[reg] = ((mpu->MAIR[reg] & ~mask) | (val & mask));
+
+  mpu->MAIR[reg] = ((mpu->MAIR[reg] & ~mask) | ((attr << pos) & mask));
 }
 
 /** Set the memory attribute encoding.
-* \param idx The attribute index to be set [0-7]
-* \param attr The attribute value to be set.
-*/
+ * \param idx The attribute index to be set [0-7]
+ * \param attr The attribute value to be set.
+ */
 __STATIC_INLINE void ARM_MPU_SetMemAttr(uint8_t idx, uint8_t attr)
 {
   ARM_MPU_SetMemAttrEx(MPU, idx, attr);
@@ -281,28 +301,28 @@
 
 #ifdef MPU_NS
 /** Set the memory attribute encoding to the Non-secure MPU.
-* \param idx The attribute index to be set [0-7]
-* \param attr The attribute value to be set.
-*/
+ * \param idx The attribute index to be set [0-7]
+ * \param attr The attribute value to be set.
+ */
 __STATIC_INLINE void ARM_MPU_SetMemAttr_NS(uint8_t idx, uint8_t attr)
 {
   ARM_MPU_SetMemAttrEx(MPU_NS, idx, attr);
 }
-#endif
+#endif /* MPU_NS */
 
 /** Clear and disable the given MPU region of the given MPU.
-* \param mpu Pointer to MPU to be used.
-* \param rnr Region number to be cleared.
-*/
-__STATIC_INLINE void ARM_MPU_ClrRegionEx(MPU_Type* mpu, uint32_t rnr)
+ * \param mpu Pointer to MPU to be used.
+ * \param rnr Region number to be cleared.
+ */
+__STATIC_INLINE void ARM_MPU_ClrRegionEx(MPU_Type *mpu, uint32_t rnr)
 {
   mpu->RNR = rnr;
   mpu->RLAR = 0U;
 }
 
 /** Clear and disable the given MPU region.
-* \param rnr Region number to be cleared.
-*/
+ * \param rnr Region number to be cleared.
+ */
 __STATIC_INLINE void ARM_MPU_ClrRegion(uint32_t rnr)
 {
   ARM_MPU_ClrRegionEx(MPU, rnr);
@@ -310,21 +330,22 @@
 
 #ifdef MPU_NS
 /** Clear and disable the given Non-secure MPU region.
-* \param rnr Region number to be cleared.
-*/
+ * \param rnr Region number to be cleared.
+ */
 __STATIC_INLINE void ARM_MPU_ClrRegion_NS(uint32_t rnr)
-{  
+{
   ARM_MPU_ClrRegionEx(MPU_NS, rnr);
 }
-#endif
+#endif /* MPU_NS */
 
 /** Configure the given MPU region of the given MPU.
-* \param mpu Pointer to MPU to be used.
-* \param rnr Region number to be configured.
-* \param rbar Value for RBAR register.
-* \param rlar Value for RLAR register.
-*/   
-__STATIC_INLINE void ARM_MPU_SetRegionEx(MPU_Type* mpu, uint32_t rnr, uint32_t rbar, uint32_t rlar)
+ * \param mpu Pointer to MPU to be used.
+ * \param rnr Region number to be configured.
+ * \param rbar Value for RBAR register.
+ * \param rlar Value for RLAR register.
+ */
+__STATIC_INLINE void ARM_MPU_SetRegionEx(MPU_Type *mpu, uint32_t rnr,
+                                         uint32_t rbar, uint32_t rlar)
 {
   mpu->RNR = rnr;
   mpu->RBAR = rbar;
@@ -332,93 +353,107 @@
 }
 
 /** Configure the given MPU region.
-* \param rnr Region number to be configured.
-* \param rbar Value for RBAR register.
-* \param rlar Value for RLAR register.
-*/   
-__STATIC_INLINE void ARM_MPU_SetRegion(uint32_t rnr, uint32_t rbar, uint32_t rlar)
+ * \param rnr Region number to be configured.
+ * \param rbar Value for RBAR register.
+ * \param rlar Value for RLAR register.
+ */
+__STATIC_INLINE void ARM_MPU_SetRegion(uint32_t rnr, uint32_t rbar,
+                                       uint32_t rlar)
 {
   ARM_MPU_SetRegionEx(MPU, rnr, rbar, rlar);
 }
 
 #ifdef MPU_NS
 /** Configure the given Non-secure MPU region.
-* \param rnr Region number to be configured.
-* \param rbar Value for RBAR register.
-* \param rlar Value for RLAR register.
-*/   
-__STATIC_INLINE void ARM_MPU_SetRegion_NS(uint32_t rnr, uint32_t rbar, uint32_t rlar)
+ * \param rnr Region number to be configured.
+ * \param rbar Value for RBAR register.
+ * \param rlar Value for RLAR register.
+ */
+__STATIC_INLINE void ARM_MPU_SetRegion_NS(uint32_t rnr, uint32_t rbar,
+                                          uint32_t rlar)
 {
-  ARM_MPU_SetRegionEx(MPU_NS, rnr, rbar, rlar);  
+  ARM_MPU_SetRegionEx(MPU_NS, rnr, rbar, rlar);
 }
-#endif
+#endif /* MPU_NS */
 
-/** Memcpy with strictly ordered memory access, e.g. used by code in ARM_MPU_LoadEx()
-* \param dst Destination data is copied to.
-* \param src Source data is copied from.
-* \param len Amount of data words to be copied.
-*/
-__STATIC_INLINE void ARM_MPU_OrderedMemcpy(volatile uint32_t* dst, const uint32_t* __RESTRICT src, uint32_t len)
+/** Memcpy with strictly ordered memory access, e.g. used by code in
+ * ARM_MPU_LoadEx()
+ * \param dst Destination data is copied to.
+ * \param src Source data is copied from.
+ * \param len Amount of data words to be copied.
+ */
+__STATIC_INLINE void ARM_MPU_OrderedMemcpy(volatile uint32_t *dst,
+                                           const uint32_t *__RESTRICT src,
+                                           uint32_t len)
 {
   uint32_t i;
-  for (i = 0U; i < len; ++i) 
+  for (i = 0U; i < len; ++i)
   {
     dst[i] = src[i];
   }
 }
 
 /** Load the given number of MPU regions from a table to the given MPU.
-* \param mpu Pointer to the MPU registers to be used.
-* \param rnr First region number to be configured.
-* \param table Pointer to the MPU configuration table.
-* \param cnt Amount of regions to be configured.
-*/
-__STATIC_INLINE void ARM_MPU_LoadEx(MPU_Type* mpu, uint32_t rnr, ARM_MPU_Region_t const* table, uint32_t cnt) 
+ * \param mpu Pointer to the MPU registers to be used.
+ * \param rnr First region number to be configured.
+ * \param table Pointer to the MPU configuration table.
+ * \param cnt Amount of regions to be configured.
+ */
+__STATIC_INLINE void ARM_MPU_LoadEx(MPU_Type *mpu, uint32_t rnr,
+                                    ARM_MPU_Region_t const *table, uint32_t cnt)
 {
-  const uint32_t rowWordSize = sizeof(ARM_MPU_Region_t)/4U;
-  if (cnt == 1U) {
+  const uint32_t rowWordSize = sizeof(ARM_MPU_Region_t) / 4U;
+  if (cnt == 1U)
+  {
     mpu->RNR = rnr;
     ARM_MPU_OrderedMemcpy(&(mpu->RBAR), &(table->RBAR), rowWordSize);
-  } else {
-    uint32_t rnrBase   = rnr & ~(MPU_TYPE_RALIASES-1U);
+  }
+  else
+  {
+    uint32_t rnrBase = rnr & ~(MPU_TYPE_RALIASES - 1U);
     uint32_t rnrOffset = rnr % MPU_TYPE_RALIASES;
-    
+
     mpu->RNR = rnrBase;
-    while ((rnrOffset + cnt) > MPU_TYPE_RALIASES) {
+    while ((rnrOffset + cnt) > MPU_TYPE_RALIASES)
+    {
       uint32_t c = MPU_TYPE_RALIASES - rnrOffset;
-      ARM_MPU_OrderedMemcpy(&(mpu->RBAR)+(rnrOffset*2U), &(table->RBAR), c*rowWordSize);
+      ARM_MPU_OrderedMemcpy(&(mpu->RBAR) + (rnrOffset * 2U), &(table->RBAR),
+                            c * rowWordSize);
       table += c;
       cnt -= c;
       rnrOffset = 0U;
       rnrBase += MPU_TYPE_RALIASES;
       mpu->RNR = rnrBase;
     }
-    
-    ARM_MPU_OrderedMemcpy(&(mpu->RBAR)+(rnrOffset*2U), &(table->RBAR), cnt*rowWordSize);
+
+    ARM_MPU_OrderedMemcpy(&(mpu->RBAR) + (rnrOffset * 2U), &(table->RBAR),
+                          cnt * rowWordSize);
   }
 }
 
 /** Load the given number of MPU regions from a table.
-* \param rnr First region number to be configured.
-* \param table Pointer to the MPU configuration table.
-* \param cnt Amount of regions to be configured.
-*/
-__STATIC_INLINE void ARM_MPU_Load(uint32_t rnr, ARM_MPU_Region_t const* table, uint32_t cnt) 
+ * \param rnr First region number to be configured.
+ * \param table Pointer to the MPU configuration table.
+ * \param cnt Amount of regions to be configured.
+ */
+__STATIC_INLINE void ARM_MPU_Load(uint32_t rnr, ARM_MPU_Region_t const *table,
+                                  uint32_t cnt)
 {
   ARM_MPU_LoadEx(MPU, rnr, table, cnt);
 }
 
 #ifdef MPU_NS
 /** Load the given number of MPU regions from a table to the Non-secure MPU.
-* \param rnr First region number to be configured.
-* \param table Pointer to the MPU configuration table.
-* \param cnt Amount of regions to be configured.
-*/
-__STATIC_INLINE void ARM_MPU_Load_NS(uint32_t rnr, ARM_MPU_Region_t const* table, uint32_t cnt) 
+ * \param rnr First region number to be configured.
+ * \param table Pointer to the MPU configuration table.
+ * \param cnt Amount of regions to be configured.
+ */
+__STATIC_INLINE void ARM_MPU_Load_NS(uint32_t rnr,
+                                     ARM_MPU_Region_t const *table,
+                                     uint32_t cnt)
 {
   ARM_MPU_LoadEx(MPU_NS, rnr, table, cnt);
 }
-#endif
-
-#endif
+#endif /* MPU_NS */
 
+#endif /* !ARM_MPU_ARMV8_H */
