# SPDX-License-Identifier: Apache-2.0 OR ISC OR MIT
--- /tmp/system_ARMCM33.c.orig	2026-01-13 17:06:53.809238718 +0000
+++ envs/m33-an524/src/platform/system_ARMCM33.c	2026-01-13 15:11:08.435799683 +0000
@@ -43,6 +43,8 @@
   #error device not specified!
 #endif
 
+#include "uart.h"
+
 /*----------------------------------------------------------------------------
   Define clocks
  *----------------------------------------------------------------------------*/
@@ -94,4 +96,36 @@ void SystemInit (void)
 #endif
 
   SystemCoreClock = SYSTEM_CLOCK;
+  uart_init();
+}
+
+/*----------------------------------------------------------------------------
+  Wrapped syscalls for newlib stdio - use semihosting for output
+ *----------------------------------------------------------------------------*/
+#include <errno.h>
+#undef errno
+extern int errno;
+
+int __wrap__read(int file, char *ptr, int len)
+{
+  (void)file;
+  (void)ptr;
+  (void)len;
+  errno = ENOSYS;
+  return -1;
+}
+
+int __wrap__write(int file, char *ptr, int len)
+{
+  if (file == 1 || file == 2)
+  { /* stdout or stderr */
+    /* Write each character using UART */
+    for (int i = 0; i < len; ++i)
+    {
+      uart_putc(ptr[i]);
+    }
+    return len;
+  }
+  errno = EBADF;
+  return -1;
 }
